internal/server/server.go

```diff
@@ -107,9 +109,6 @@ type Addrs struct {
 
 // newServer creates a Server with base dependencies initialized to zero values.
 func newServer(options Options) *Server {
-	if options.Hostname == "" {
```

Comment:
I think it is better to have some default here. Maybe localhost?

Reply:
I think this is the wrong place to set defaults, they should go into `defaultServerOptions`. 

What purpose does this default serve? I was going to move it to `defaultServerOptions` until I realized no tests were failing, and I couldn't find any reason to set a default. 

If we have a reason for a default I can add a comment about it, but in general I think we are better off trying to make things work with a zero value default, or to at least know exactly why we can't use a zero value default.

Reply:
My guess was just that it's easier to see that you need to set a hostname when the base URL becomes example.com, but you're right that it doesn't match other places

Reply:
We discussed this a bit in slack recently. I think #3072 is a better option for setting a default org domain, instead of relying on a default in code that will never match production.